server/src/main/java/org/elasticsearch/action/admin/cluster/stats/MappingStats.java

```diff
@@ -62,15 +69,28 @@ public static MappingStats of(Metadata metadata, Runnable ensureNotCancelled) {
                 continue;
             }
             AnalysisStats.countMapping(mappingCounts, indexMetadata);
+
+            var sourceMode = SourceFieldMapper.INDEX_MAPPER_SOURCE_MODE_SETTING.get(indexMetadata.getSettings());
+            indexSourceModeUsageCount.compute(sourceMode.toString().toLowerCase(Locale.ENGLISH), (k, v) -> v == null ? 1 : v + 1);
         }
         final AtomicLong totalFieldCount = new AtomicLong();
         final AtomicLong totalDeduplicatedFieldCount = new AtomicLong();
+        // Account different source modes based on _source.mode mapping attribute in a separate map:
+        // (Materializing the mapping into map of maps is expensive and only happens here once per unique mapping)
+        Map<String, Integer> mappingSourceModeUsageCount = new HashMap<>();
         for (Map.Entry<MappingMetadata, Integer> mappingAndCount : mappingCounts.entrySet()) {
             ensureNotCancelled.run();
             Set<String> indexFieldTypes = new HashSet<>();
             Set<String> indexRuntimeFieldTypes = new HashSet<>();
             final int count = mappingAndCount.getValue();
             final Map<String, Object> map = mappingAndCount.getKey().getSourceAsMap();
+            if (map.containsKey("_source")) {
+                Map<?, ?> sourceFieldDefinition = (Map<?, ?>) map.get("_source");
+                if (sourceFieldDefinition.containsKey("mode")) {
+                    String mode = (String) sourceFieldDefinition.get("mode");
+                    mappingSourceModeUsageCount.compute(mode.toString().toLowerCase(Locale.ENGLISH), (k, v) -> v == null ? 1 : v + 1);
```

Comment:
Is it possible to account for indices with _source.mode specified twice?

Reply:
Here we iterate over all unique `MappingMetadata` instances, so I don't think we see same mappings multiple times?  So I don't we account same indices multiple times in `mappingSourceModeUsageCount`? We can under account, when mappings are identical between indices.

If both `index.mappings.source.mode` index setting and `_source.mode` mapping attribute is used then this change picks either the source mode stats or index mode stats. 

I'm also ok with just accounting source mode from `index.mappings.source.mode` index setting, since _source.mode is on its way out and only tech preview synthetic source usages are configured that way. Maybe account `_source.mode` usages is not worth the complexity?

Reply:
> Maybe account _source.mode usages is not worth the complexity?
++